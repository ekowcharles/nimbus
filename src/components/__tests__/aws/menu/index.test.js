import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Menu from '../../../aws/menu/index'

const menuOptions = {
  "az": {
    "hideName": false,
  },
  "bucket": {
    "hide": false,
    "hideName": false,
  },
  "database": {
    "hide": false,
    "hideName": false,
    "hideStatus": false,
  },
  "instance": {
    "hide": false,
    "hideStatus": false,
    "hideType": false,
  },
  "subnet": {
    "hideCidr": false,
    "hideId": false,
    "hideStatus": false,
  },
  "vpc": {
    "hideCidr": false,
    "hideIcon": false,
    "hideId": false,
    "hideStatus": false,
  },
}

describe('Menu', () => {
  it('renders menu', () => {
    const wrapper = shallow(
      <Menu
        menuOptions={menuOptions}
      />
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})