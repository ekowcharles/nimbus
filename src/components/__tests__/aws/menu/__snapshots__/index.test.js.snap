// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders menu 1`] = `
<div
  className="menu"
>
  <ul>
    <li
      className="dropdown"
    >
      <span
        className="dropbtn"
      >
        VPC
      </span>
      <div
        className="dropdown-content"
      >
        <MenuItem
          checked={false}
          label="Hide Icon"
          update={[Function]}
        />
        <MenuItem
          checked={false}
          label="Hide Id"
          update={[Function]}
        />
        <MenuItem
          checked={false}
          label="Hide Cidr"
          update={[Function]}
        />
        <MenuItem
          checked={false}
          label="Hide Status"
          update={[Function]}
        />
      </div>
    </li>
    <li
      className="dropdown"
    >
      <span
        className="dropbtn"
      >
        Subnet
      </span>
      <div
        className="dropdown-content"
      >
        <MenuItem
          checked={false}
          label="Hide Id"
          update={[Function]}
        />
        <MenuItem
          checked={false}
          label="Hide Cidr"
          update={[Function]}
        />
        <MenuItem
          checked={false}
          label="Hide Status"
          update={[Function]}
        />
      </div>
    </li>
    <li
      className="dropdown"
    >
      <span
        className="dropbtn"
      >
        AZ
      </span>
      <div
        className="dropdown-content"
      >
        <MenuItem
          checked={false}
          label="Hide Name"
          update={[Function]}
        />
      </div>
    </li>
    <li
      className="dropdown"
    >
      <span
        className="dropbtn"
      >
        Instance
      </span>
      <div
        className="dropdown-content"
      >
        <MenuItem
          checked={false}
          label="Hide"
          update={[Function]}
        />
        <MenuItem
          checked={false}
          label="Hide Status"
          update={[Function]}
        />
        <MenuItem
          checked={false}
          label="Hide Type"
          update={[Function]}
        />
      </div>
    </li>
    <li
      className="dropdown"
    >
      <span
        className="dropbtn"
      >
        Database
      </span>
      <div
        className="dropdown-content"
      >
        <MenuItem
          checked={false}
          label="Hide"
          update={[Function]}
        />
        <MenuItem
          checked={false}
          label="Hide Name"
          update={[Function]}
        />
        <MenuItem
          checked={false}
          label="Hide Status"
          update={[Function]}
        />
      </div>
    </li>
    <li
      className="dropdown"
    >
      <span
        className="dropbtn"
      >
        Bucket
      </span>
      <div
        className="dropdown-content"
      >
        <MenuItem
          checked={false}
          label="Hide"
          update={[Function]}
        />
        <MenuItem
          checked={false}
          label="Hide Name"
          update={[Function]}
        />
      </div>
    </li>
  </ul>
</div>
`;
